File: spring-jpa-projection/src/main/java/com/jpa/projection/controller/user/UserController.java
package com.jpa.projection.controller.user;

import com.jpa.projection.service.user.UserService;
import com.jpa.projection.service.user.response.UserPostResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/users")
public class UserController {

    private final UserService userService;

    @GetMapping("/{id}/post")
    public UserPostResponse getUserPost(@PathVariable("id") final Long userId) {
        return userService.getUserPost(userId);
    }
}

File: spring-exposed-orm/src/main/kotlin/com/jjlee/exposed/controller/UserController.kt
package com.jjlee.exposed.controller

import com.jjlee.exposed.dto.request.UserCreateRequest
import com.jjlee.exposed.dto.response.UserResponse
import com.jjlee.exposed.service.UserService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(
    private val userService: UserService
) {

    @GetMapping("/users/{id}")
    fun getUser(@PathVariable id: Long) : UserResponse {
        return userService.findUserById(id)
    }

    @PostMapping("/users")
    fun createUser(@RequestBody request: UserCreateRequest) : Long {
        return userService.create(request)
    }

    @DeleteMapping("/users/{id}")
    fun deleteUser(@PathVariable id: Long) {
        userService.delete(id)
    }
}